internal/infrastructure/kubernetes/infra.go

```diff
@@ -66,6 +67,10 @@ func (i *Infra) createOrUpdate(ctx context.Context, r ResourceRender) error {
 		return fmt.Errorf("failed to create or update deployment %s/%s: %w", i.Namespace, r.Name(), err)
 	}
 
+	if err := i.createOrUpdateDaemonSet(ctx, r); err != nil {
+		return fmt.Errorf("failed to create or update daemonset %s/%s: %w", i.Namespace, r.Name(), err)
+	}
+
```

Comment:
note: some of the resources may not be relevant for Daemonset. For example, HPAs cannot be used for Daemonsets: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/, so maybe we should not create it in the DS flow.

Reply:
You're right! I completely forgot about that. I will add CEL validation and ignore (delete) HPA if DaemonSet is used.